src/rer/solrpush/adapters/file.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+from rer.solrpush.interfaces.adapter import IExtractFileFromTika
+from zope.interface import implementer
+
+
+@implementer(IExtractFileFromTika)
+class FileExtractor(object):
+    def __init__(self, context):
+        self.context = context
+
+    def get_file_to_index(self):
+        """
+        """
+        if not self.context.file:
+            return ""
+        form = self.context.REQUEST.form
```

Comment:
getting the file from the request ? why ?

Reply:
I don't remember: probably with previous queue implementation there was a moment where i came here without the file.

I tested it right now and this doesn't happen anymore: cleaned the code in last commit